data-loader/core/src/main/java/com/scalar/db/dataloader/core/util/TableMetadataUtil.java

```diff
@@ -0,0 +1,144 @@
+package com.scalar.db.dataloader.core.util;
+
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.dataloader.core.Constants;
+import com.scalar.db.dataloader.core.dataimport.controlfile.ControlFileTable;
+import com.scalar.db.io.DataType;
+import com.scalar.db.transaction.consensuscommit.Attribute;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+
+/** Utility class for handling ScalarDB table metadata operations. */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class TableMetadataUtil {
+
+  /**
+   * Determines whether a given column is a metadata column based on predefined criteria.
+   *
+   * @param columnName The name of the table column to check.
+   * @param metadataColumns A set of predefined metadata columns.
+   * @param columnNames A set of all column names in the table.
+   * @return {@code true} if the column is a metadata column; {@code false} otherwise.
+   */
+  public static boolean isMetadataColumn(
```

Comment:
You can use `ConsensusCommitUtils.isTransactionMetaColumn()`:
https://github.com/scalar-labs/scalardb/blob/98b5e1b737a5ceab9269c28c7ac0991e3a3e13f8/core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitUtils.java#L201